week2/index.js

```diff
@@ -1,65 +1,129 @@
 }
 
+//Call help file and display content.
 function showHelp() {
-  openFile('help.txt', function(error, data) {
-    if (error) {
-      return console.log('Error: the help file could not be displayed', error);
-    }
-    console.log(data);
-  });
+	var data = openFile('help.txt')
+	console.log(data);
 }
 
+//call todo file content and display its content.
 function listTodos() {
-  openFile('todo.txt', function(error, data) {
-    if (error) {
-      if (error.code === 'ENOENT') {
-        return console.log('Nothing to do! (or your dog ate your todo list)');
-      } else {
-        return console.log('Error: Something went wrong', error);
-      }
-    }
+	var data = openFile('todo.txt');
+	var todos = splitStringByNewline(data);
+	if (todos.length === 0) {
+		return console.log('Nothing to do!')
+	}
+	console.log('Your todo list looks like this');
+	todos.forEach(function (element, index) {
+		index = (index + 1).toString();
+		console.log(index, element);
+	});
+
+	if (todos.length > 5) {
+		console.log('You have too much to do!');
+	}
+
+}
+//Add new todo line, creat a file in case its not exist.
+function addLine() {
+	options.shift(); //remove first element from command
+	var allLine = options.join(' ') ; //wrap all command as one line
+	fs.appendFileSync('todo.txt', allLine + '\n', 'utf8'); // append the new line to list
+	listTodos(); // show the new list
+}
 
-    var todos = splitStringByNewline(data);
+//update new todo line, creat a file in case its not exist.
+function updateLine() {
+	if (options[1] <= 0) { // in case that the user enter worng number
+		console.log('Please enter right line number!!! ' + '\n');
+		showHelp();
+	} else {
+		var lineNumber = options[1] - 1; // get the line number
+		var data = fs.readFileSync('todo.txt', 'utf8') // get file information
+		var dataRendering = data.split('\n'); //split every line to array element
+		if (lineNumber > dataRendering.length) { // check if update line number exist or not
+			console.log("You dont have this line number in your list\n" );
+			showHelp();
+		} else {
+			var updateLine = options.slice(2); // remove first two words from command 
+			dataRendering.splice(lineNumber, 1, updateLine.join(' ')); // replace with index number for length of one with update line.
+			var newLines = dataRendering.join('\n'); // join array element with new line separator
+			fs.writeFileSync('todo.txt', newLines); //Write on file
+			console.log('Line ' + ++lineNumber + ' is Updated!' + '\n');// display confirmation message
+			listTodos(); // show the new list
+		}
+	}
 
-    if (todos.length === 0) {
-      return console.log('Nothing to do!')
-    }
+}
 
-    console.log('Your todo list looks like this');
-    todos.forEach(function(element, index) {
-      index = (index + 1).toString();
-      console.log(index, element);
-    });
+//delete one line from todo list by index number.
+function deleteLine() {
+	if (options[1] <= 0) { //Check if the line number is 0 and show help
```

Comment:
What happens if the user has 4 items and enters `5` or greater as the line number? How could we check for that?